### PR TITLE
[SOL-1831] Show maximum uses in coupon detail page

### PR DESCRIPTION
**Pull request by iivic**
@marjev @mjfrey @vkaracic  Someone please review.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @iivic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=812&repo=edx%2Fecommerce

File: ecommerce/static/js/test/specs/views/coupon_detail_view_spec.js [hunk 1]
```diff
@@ -152,6 +152,14 @@ define([
                 expect(e.preventDefault).toHaveBeenCalled();
                 expect(window.open).toHaveBeenCalledWith(url, '_blank');
             });
+
```
Comment (vkaracic) [inline ecommerce/static/js/test/specs/views/coupon_detail_view_spec.js:187]: Try including `max_uses` in the model used in test on line 81 in this spec, and a line something like:

```
expect(view.$el.find('.max-uses > .value').text()).toEqual(model.get('max_uses'));
```

In your test below you are testing your function, but nowhere are you testing if that function is called on render.

**Comment by vkaracic** [comment]
After the one comment is addressed :+1:

File: ecommerce/static/js/views/coupon_detail_view.js [hunk 2]
```diff
@@ -66,6 +66,14 @@ define([
                 return '';
             },
 
+            setMaxUses: function(voucher) {
+               if (voucher.usage === 'Single use') {
+                   return 1;
```
Comment (marjev) [inline ecommerce/static/js/views/coupon_detail_view.js]: Change this in the model. So, if voucher.usage is 'Single use', set model max_uses to 1. This way, we let model handle the logic and we display model's data on the template.

File: ecommerce/static/js/views/coupon_detail_view.js [hunk 3]
```diff
@@ -84,6 +92,7 @@ define([
                     usage: this.usageLimitation(voucher),
                     category: category,
                     note: note,
+                    maxUses: this.setMaxUses(voucher)
```
Comment (marjev) [inline ecommerce/static/js/views/coupon_detail_view.js]: Once you address this comment https://github.com/edx/ecommerce/pull/812/files#r67881116, you won't need this line.

**Comment by marjev** [comment]
Please provide a screenshot of the details page so that we can see the SASS/template changes.

**Comment by iivic** [comment]
Comments addressed. Here is the ss screen shot 2016-06-23 at 13 24 42

**Comment by marjev** [comment]
👍

**Comment by vkaracic** [comment]
:+1:

**Pull request merged**